### DISCUSSION TITLE
Get list of items having a specific label

### DISCUSSION THREAD

**Discussion by omarmallat** [pnp/powershell#2372] [Q&A]
I'm trying set a specific label on all items in a specific folder in a SharePoint Site...
Actually most of the items already have this label.. only few items have the wrong label...
So instead of looping through all items in a folder, which is taking like 24 hours, I prefer to get first a list of the items who have wrong label, and use the command 
`Set-PnPListItem -List Documents -Identity $file.ID -Label $Label
`
so what is the right command to get a list of items who have a specific label?

currently I'm using the following
```
$Label  = "No Retention"
$sourceFolder = "*/Shared Documents/CP033/Riders & Artists/*"
    $List = Get-PnPList -Identity "Documents"
  
    #Get all Items from the Library - with progress bar
    $global:counter = 0
    $ListItems = Get-PnPListItem -List $List -PageSize 500 -Fields ID -ScriptBlock { Param($items) $global:counter += $items.Count; Write-Progress -PercentComplete `
                ($global:Counter / ($List.ItemCount) * 100) -Activity "Getting Items from Site:" -Status "Processing Items $global:Counter to $($List.ItemCount)";} | Where-Object {$_.FileSystemObjectType -contains "File" -and $_.FieldValues.FileRef -like $sourceFolder}
    Write-Progress -Activity "Completed Retrieving Files and Folders from Site!" -Completed


$files = $ListItems  | Where-Object {$_.FileSystemObjectType -contains "File" -and $_.FieldValues.FileRef -like $sourceFolder}

```

**Comment by kmloftin** [comment]
I used this example and it seems that it is what you're looking for. https://getsecureandcompliant.com/use-powershell-to-set-a-retention-label-on-an-item-in-sharepoint-online/

**Reply by ryszardcopa** [reply]
This link is not working for me, I am also interested in it.
